File: 0540-single-element-in-a-sorted-array/0540-single-element-in-a-sorted-array.go
func singleNonDuplicate(nums []int) int {
    n:=len(nums)
    low:=0
    high:=n-1

    for low<high{
        guess:=(low+high)/2

        if guess%2==1{
            guess--
        }

        if nums[guess]==nums[guess+1]{
            low=guess+2
        }else{
            high=guess
        }
    }
    return nums[low]
}

File: 0540-single-element-in-a-sorted-array/0540-single-element-in-a-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var singleNonDuplicate = function(nums) {
    let n=nums.length;
    let low=0;
    let high=n-1;

    while(low<high){
        let guess=Math.floor((low+high)/2);

        if(guess%2==1){
            guess--;
        }

        if(nums[guess]==nums[guess+1]){
            low=guess+2;
        }
        else{
            high=guess;
        }
    }
    return nums[low];
};